osgpkitools/incommon_request.py

```diff
@@ -60,172 +62,97 @@
-        default=[]
-    )
-    parser.add_option(
-        '-u',
-        '--username',
-        action='store',
-        dest='login',
-        help='Provide the InCommon username (login).',
-        metavar='LOGIN',
-        default=[]
-    )
-    parser.add_option(
-        '-T',
-        '--test',
-        action='store_true',
-        dest='test',
-        help='Test connection to InCommon API. Useful to test authentication credentials',
-        default=False
-    )
-    parser.add_option(
-        '',
-        '--debug',
-        dest='debug',
-        help="Write debug output to stdout",
-        action='store_true',
-        default=False
-    )
+    '''%(prog)s [--debug] -u username -k pkey -c cert \\
+           (-H hostname | -F hostfile) [-a altnames] [-d write_directory]
+       %(prog)s [--debug] -u username -k pkey -c cert -t
+       %(prog)s -h
+       %(prog)s --version'''
+ 
+    parser = argparse.ArgumentParser(add_help=False, usage=usage, 
+                                     description='Request and retrieve certificates from the  InCommon IGTF server CA.')
+
+    required = parser.add_argument_group('Required', 'Specify only one of -H/--hostname and -F/--hostfile')
+    hosts = required.add_mutually_exclusive_group()
+    hosts.add_argument('-H', '--hostname', action='store', dest='hostname',
+                        help='The hostname (FQDN) to request. If specified, -F/--hostfile will be ignored')
+    hosts.add_argument('-F', '--hostfile', action=FilePathAction, dest='hostfile',
+                       help='File containing list of hostnames (FQDN), one per line, to request. Space separated '
+                       'subject alternative names (SANs) may be specified on the same line as each hostname.')
+
+    required.add_argument('-u', '--username', action='store', required=True, dest='login',
+                          help="Specify requestor's InCommon username/login")
+
+    required.add_argument('-c', '--cert', action=FilePathAction, required=True, dest='usercert',
+                          help="Specify requestor's user certificate (PEM Format)")
     
-    parser.add_option_group(group)
-    (args, values) = parser.parse_args()
-
-    if args.debug:
+    required.add_argument('-k', '--pkey', action=FilePathAction, required=True, dest='userprivkey',
+                          help="Specify requestor's private key (PEM Format)")
+
+    optional = parser.add_argument_group("Optional")
+    optional.add_argument('-h', '--help', action='help',
+                          help='show this help message and exit')
+    optional.add_argument('-a', '--altname', action='append', dest='altnames', default=[],
+                          help='Specify the SAN for the requested certificate (only works with -H/--hostname). '
+                          'May be specified more than once for additional SANs.')
+    optional.add_argument('-o', '--outputdir', action='store', dest='write_directory', default='.',
```

Comment:
"-o" does not match the usage.  It used to be -d, did you intend to change it?  I think it would be better to match the usage of osg-gridadmin-cert-request and osg-cert-request.

Reply:
I changed it to -d, but I will return it to -d to match with osg-cert-request.